src/main/java/yapper/exceptions/ExceptionHandler.java

```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+
+    // File Input Parsing Exceptions
+
+    /**
+     * Validates if the specified task type is valid when reading from file.
+     *
+     * @param taskType the type of the task
+     * @throws YapperException if the task type is invalid
+     */
+    public static void checkIfTaskTypeValid(String taskType) throws YapperException {
+        if ( !taskType.equals(StringStorage.TODO_SYMBOL) &&
+             !taskType.equals(StringStorage.DEADLINE_SYMBOL) &&
+             !taskType.equals(StringStorage.EVENT_SYMBOL) ) {
+            throw new YapperException(StringStorage.INVALID_TASK_TYPE_MESSAGE);
+        }
+    }
+    /**
+     * Validates if the specified task status is valid when reading from file.
+     *
+     * @param taskStatus the status of the task
+     * @throws YapperException if the task status is invalid
+     */
+    public static void checkIfTaskStatusValid(String taskStatus) throws YapperException {
+        if (!taskStatus.equals(StringStorage.IS_DONE_SYMBOL) &&
+            !taskStatus.equals(StringStorage.NOT_DONE_SYMBOL)) {
+            throw new YapperException(StringStorage.INVALID_TASK_STATUS_MESSAGE);
+        }
+    }
+
+
+    // Exceptions for: DELETE, MARK, UNMARK
+
+    /**
+     * Checks if there are tasks in the list before performing DELETE, MARK, or UNMARK operations.
+     *
+     * @param currTaskTotal the total number of tasks currently available
+     * @throws YapperException if the task list is empty
+     */
+    public static void checkIfTaskOrdinalIsOutOfRange(int currTaskTotal) throws YapperException {
+        if (currTaskTotal == 0) {
+            throw new YapperException(StringStorage.LIST_EMPTY_MESSAGE);
+        }
+    }
+    /**
+     * Validates if the specified task ordinal is within the valid range of tasks.
+     *
+     * @param currTaskTotal the total number of tasks currently available
+     * @param taskOrdinal the ordinal of the task to validate
+     * @throws YapperException if the task ordinal is out of range
+     */
+    public static void checkIfTaskOrdinalIsOutOfRange(int currTaskTotal, int taskOrdinal) throws YapperException {
+        if (taskOrdinal < 0 || taskOrdinal >= currTaskTotal) {
+            throw new YapperException(StringStorage.LIST_OOB_MESSAGE);
+        }
+    }
+    /**
+     * Checks if the done status needs to be changed based on the old and new status.
+     *
+     * @param oldStatus the current status of the task
+     * @param newStatus the new status to be applied
+     * @throws YapperException if trying to change to the same status
+     */
+    public static void checkIfDoneStatusNeedsChanging(boolean oldStatus, boolean newStatus) throws YapperException {
+        if (oldStatus && newStatus) {
+            throw new YapperException(StringStorage.TASK_ALREADY_DONE_MESSAGE);
+        } else if (!oldStatus && !newStatus) {
+            throw new YapperException(StringStorage.TASK_STILL_NOT_DONE_MESSAGE);
+        }
+    }
+
+}
```

Comment:
can consider splitting it to 2 classes one for reading data from storage other for managing user inputs